flink-contrib/flink-storm-examples/src/main/java/org/apache/flink/storm/util/BoltFileSink.java

```diff
@@ -18,20 +18,23 @@
 package org.apache.flink.storm.util;
 
 import backtype.storm.task.TopologyContext;
+import org.apache.flink.core.fs.FSDataOutputStream;
+import org.apache.flink.core.fs.FileSystem;
+import org.apache.flink.core.fs.Path;
 
 import java.io.BufferedWriter;
-import java.io.FileWriter;
 import java.io.IOException;
+import java.io.OutputStreamWriter;
 import java.util.Map;
 
 /**
- * Implements a sink that write the received data to the given file (as a result of {@code Object.toString()} for each
+ * Implements a sink that writes the received data to the given file (as a result of {@code Object.toString()} for each
  * attribute).
  */
 public final class BoltFileSink extends AbstractBoltSink {
 	private static final long serialVersionUID = 2014027288631273666L;
 
-	private final String path;
+	private final Path path;
 	private BufferedWriter writer;
```

Comment:
No. People will have spout/bolt code they do not want to touch when running it in Flink. Thus there code will be written in the same way -- and so should be example be. Otherwise, we deliver the impression they need to change there code -- but they don't. Thus, we implement the example Spout/Bolts in a pure Storm way. Of course, if somebody develops a new Spout/Bolt with Flink in mind, your approach makes sense. However, this is not the main focus (it would be even better if the code new stuff Flink native in embedded mode, instead of develop Spout/Bolts which are Flink tailored).